### PR TITLE
Enhancement/6221 - Conditionally render GA4 metrics AdminBar widgets

### PR DESCRIPTION
**Pull request by sashadoes**
## Summary

Addresses issue:

- #6221 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/adminbar/AdminBarWidgets.js [hunk 1]
```diff
@@ -28,31 +28,45 @@ import Data from 'googlesitekit-data';
...
 	withWidgetComponentProps( WIDGET_IMPRESSIONS )( AdminBarImpressions );
 const AdminBarClicksWidget =
 	withWidgetComponentProps( WIDGET_CLICKS )( AdminBarClicks );
+
 // Analytics Widgets.
 const AdminBarUniqueVisitorsWidget = withWidgetComponentProps(
 	WIDGET_VISITORS
 )( AdminBarUniqueVisitors );
 const AdminBarSessionsWidget =
 	withWidgetComponentProps( WIDGET_SESSIONS )( AdminBarSessions );
```
Comment (techanvil) [inline assets/js/components/adminbar/AdminBarWidgets.js:58]: I realise it's a divergence from the IB, but we could simplify this change and also make it more efficient by taking the following approach instead, and removing the rendering changes. I'd suggest we go with this, what do you think?

```suggestion
const ga4ReportingEnabled = isFeatureEnabled( 'ga4Reporting' );

// Analytics Widgets.
const AdminBarUniqueVisitorsWidget = withWidgetComponentProps(
	WIDGET_VISITORS
)( ga4ReportingEnabled ? AdminBarUniqueVisitors : AdminBarUniqueVisitorsGA4 );

const AdminBarSessionsWidget = withWidgetComponentProps( WIDGET_SESSIONS )(
	ga4ReportingEnabled ? AdminBarSessions : AdminBarSessionsGA4
);
```

**Review by techanvil — CHANGES_REQUESTED** [review]
Hey @hussain-t, this is looking good. I've got a suggestion for a small improvement, please take a look.

File: assets/js/components/adminbar/AdminBarWidgets.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/components/adminbar/AdminBarWidgets.js:58]: Thanks, @techanvil, I thought the same; however, I found this method was used in our codebase. I did so to keep it consistent.
https://github.com/google/site-kit-wp/blob/3c36bc8f555ae98dcc5ac21595503661572eed88/assets/js/components/wp-dashboard/WPDashboardWidgets.js#L61-L78

Comment (hussain-t) [reply assets/js/components/adminbar/AdminBarWidgets.js:58]: Let me know WDYT?

Comment (techanvil) [reply assets/js/components/adminbar/AdminBarWidgets.js:58]: I see - thanks for pointing that out. In that case, sure, let's keep it as you've implemented it for the sake of consistency, considering as well that it does follow the IB. In the long run we'll be removing the `ga4Reporting` flag anyway and this code will be simplified then.

**Review by techanvil — APPROVED** [review]
LGTM, nice one @hussain-t!

**Pull request merged**